### ISSUE TITLE
Document Search Slow - Text

### ISSUE THREAD

**Issue by kestlund** [OregonDigital/oregondigital#804]
Within document viewer search is very slow and it is hard to tell if there is even text behind to search.  Problem record: http://oregondigital.org/catalog/oregondigital:df661w60r

If there is text, can we look into the slowness?
If not text, is there a way to notify the user that search is hopeless?

**Comment by terrellt** [comment]
Yup, documents with lots of text are going to be slow because it's searching via ruby. I've talked to @atz about how we could move it into solr, but so far I've been unable to get something figured out. The problem is we need phrase searching on a tagged document, where it should ignore the tags for search but return a result WITH the tags so that we can get the x/y coordinates on the page from it.

**Comment by srabun** [comment]
may get more info from python/newspaper (LC newspaper viewer) or IIIF work?

**Comment by jechols** [comment]
I think pdj.js can satisfy some of our requirements, maybe all of them if we're willing to get our hands slightly dirty.  For instance, here's a solution to do a search and auto-highlight-all:

```javascript
PDFFindBar.open(); //optional if you want to show the search bar
PDFFindBar.findField.value = 'your search term';
PDFFindBar.highlightAll.checked= true;
PDFFindBar.findNextButton.click();
```

(More details on search/page from URL: https://github.com/mozilla/pdf.js/issues/1875)
